### ISSUE TITLE
[Issue]: Docker defaults running jellyfin process as root

### ISSUE THREAD

**Issue by jNullj** [jellyfin/jellyfin-packaging#28]
### Please describe your bug

The jellyfin process runs as root in the docker container.
I don't have much experience but i suspect it to be bad practice, for example official mysql image for docker is checking if the container is running as root, if so it runs the process as the mysql user created at image creation. ([link to entry point script](https://github.com/docker-library/mysql/blob/37cf4047a0544046c9ecce8f2885d4f362a227bc/8.0/docker-entrypoint.sh#L376))
I belive this increases security.

### Jellyfin Version

10.7.7

### Environment

```markdown
- OS: Arch
- Virtualization: Docker
- Clients: Browser
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by joshuaboniface** [comment]
I agree with the idea here, but am not sure about the actual implementation or how it will affect e.g. existing containers.

I know at some point in the past this was tried and didn't work, but I don't know the details of what was done or why it failed. Probably HWA related and additional setup.

I'd definitely welcome PRs to address this.

**Comment by thornbill** [comment]
I'm not entirely convinced we should add any custom handling for this... running containers as root is the default in the docker ecosystem (for better or worse) but it can be provided a user or user id to run as a different user.

https://stackoverflow.com/questions/35734474/connect-to-docker-container-as-user-other-than-root#35736699

**Comment by mpeter50** [comment]
> but am not sure about the actual implementation or how it will affect e.g. existing containers

Just to give my 2 cents: an option could be to have this as a new container "flavor", like `10.9-rootless` or something like that.
If it works right, it could become the default one over time, but maybe that is not a good idea dependeng on the changes needed.

> Probably HWA related

If we can't solve that, maybe we can still have a rootless container, but with very visible signs (e.g. in the dockerhub readme, maybe also with a log line when the container is starting up) that you wont be able to use hardware acceleration (for now) with that image.

**Comment by jNullj** [comment]
We might be able to get inspiration from photoprism
[link to doc about rootless container and hwa](https://docs.photoprism.app/getting-started/advanced/docker-security/#run-services-as-non-root-user)

Can't users add permission for the user running jellyfin to the hwa device?

**Comment by joshuaboniface** [comment]
> Can't users add permission for the user running jellyfin to the hwa device?

Theoretically, yes. But HWA is already a bit of a troubleshooting nightmare and adding yet another variable into the equation is not something we're really keen on.

[There is documentation on how to run the existing container image as a non-root user](https://jellyfin.org/docs/general/installation/container) for both Docker Compose and Podman, but the default container using default docker is still root.

**Comment by ferferga** [comment]
I solved this problem in Vue this way a few weeks ago, but only for opening ports, there might be a need for extra capabilities in the server's case: https://github.com/jellyfin/jellyfin-vue/blob/master/packaging/docker/contents/postunpack.sh#L18

In case it's useful for someone to open a PR in the meantime, if not I will open it at some point since it's something I'd like to improve in the current packaging process regardless (but it can take a lot of time until I tackle it, hence why a PR is good regardless).

**Comment by briandipalma** [comment]
I'm using https://hub.docker.com/r/linuxserver/jellyfin and it has all the correct users set up from what I can tell:

<img width="1127" height="719" alt="Image" src="https://github.com/user-attachments/assets/63672683-be96-456d-9dbb-b9f7dedf5c84" />

I tried to dig into their code so see how they do this but frankly it's not clear to me. I feel the Jellyfin team should consider recommending the linuxserver image over their own as this isn't core to the jellyfin project and building docker images well can, like almost all technical endeavors, require significant knowledge.
